Source/Core/VideoCommon/BPFunctions.cpp

```diff
@@ -39,19 +39,28 @@ void SetGenerationMode()
 
 void SetScissor()
 {
-  /* NOTE: the minimum value here for the scissor rect and offset is -342.
-   * GX internally adds on an offset of 342 to both the offset and scissor
-   * coords to ensure that the register was always unsigned.
+  /* NOTE: the minimum value here for the scissor rect is -342.
+   * GX internally adds on an offset of 342 to scissor coords to
+   * ensure that the register was always unsigned.
    *
    * The code that was here before tried to "undo" this offset, but
    * since we always take the difference, the +342 added to both
    * sides cancels out. */
 
-  /* The scissor offset is always even, so to save space, the scissor offset
+  /* NOTE: With positive scissor offset the scissor rectangle is shifted left and/or up;
+   * and with negative scissor offset, the scissor rectangle is shifted right and/or down.
+   *
+   * GX internally also adds on an offset of 342 to scissor offset.
+   * The scissor offset is always even, so to save space, the scissor offset
    * register is scaled down by 2. So, if somebody calls
-   * GX_SetScissorBoxOffset(20, 20); the registers will be set to 10, 10. */
-  const int xoff = bpmem.scissorOffset.x * 2;
-  const int yoff = bpmem.scissorOffset.y * 2;
+   * GX_SetScissorBoxOffset(20, 20); the registers will be set to (10 + 171, 10 + 171).
+   *
+   * The scissor offset register is 10bit signed [-512, 511].
+   * (e.g. In Super Mario Galaxy 1 and 2, during the Boss roar effect,
+   * the scissor offset register will be set to -61, so the scissor offset is -464)
+   */
+  s32 xoff = bpmem.scissorOffset.x * 2;
+  s32 yoff = bpmem.scissorOffset.y * 2;
```

Comment:
These can still be `const`, right?

In my opinion it would be good to have two commits, one which changes the offset code and the other which changes types (including in other locations relating to scissoring, such as `native_rc` below) and adds `const` where possible (it seems like there are several places where it could be added).  But that's probably not completely necessary.

Reply:
`const int xoff = bpmem.scissorOffset.x * 2;`
xoff is just a temp variable, not a reference or point, I thought the const qualifier is probably not necessary.

Reply:
It's probably fine either way.  In most of my changes I've been changing temp variables to `const` where possible, but it's not really something that matters too much.